src/core/analytics.js

```diff
@@ -371,6 +377,98 @@ class Analytics {
     }
   }
 
+  /**
+   * A function to send single event to a destination
+   * @param {instance} destination
+   * @param {Object} rudderElement
+   * @param {String} methodName
+   */
+  sendDataToDestination(destination, rudderElement, methodName) {
+    try {
+      if (destination[methodName]) {
+        const clonedRudderElement = R.clone(rudderElement);
+        destination[methodName](clonedRudderElement);
+      }
+    } catch (err) {
+      const message = `[sendToNative]:: [Destination: ${destination.name}]:: `;
+      handleError(err, message);
+    }
+  }
+
+  /**
+   * A function to process and send events to device mode destinations
+   * @param {instance} destinations
+   * @param {Object} rudderElement
+   * @param {String} methodName
+   */
+  processAndSendEventsToDeviceMode(destinations, rudderElement, methodName) {
```

Comment:
have we tested also for events that come from localstorage retry queue?

Reply:
Local storage retry queue is only used for cloud mode events. No relation with that queue here.